commit c4d350009a43a3f789fce06ab27e30ac9b3f8857
Author: Jeffrey Bastian <jbastian@redhat.com>
Date:   Fri Jul 20 16:58:39 2012 -0500

    add RE::Init compat function for pcre-6.x
    
    patch from Ubuntu:
     https://bugs.launchpad.net/ubuntu/+source/pcre3/+bug/130428
     https://launchpad.net/ubuntu/+source/pcre3/7.2-1ubuntu2
     https://launchpadlibrarian.net/8701986/pcre3_7.2-1ubuntu2.diff.gz

diff --git a/pcrecpp.cc b/pcrecpp.cc
index e8e8a12..b9abf3b 100644
--- a/pcrecpp.cc
+++ b/pcrecpp.cc
@@ -79,6 +79,12 @@ static const string empty_string;
 // If the user doesn't ask for any options, we just use this one
 static RE_Options default_options;
 
+// keep the old version from 6.x
+void RE::Init(const char *c_pat, const RE_Options* options) {
+  const string cxx_pat(c_pat);
+  Init(cxx_pat, options);
+}
+
 void RE::Init(const string& pat, const RE_Options* options) {
   pattern_ = pat;
   if (options == NULL) {
diff --git a/pcrecpp.h b/pcrecpp.h
index a87a816..fe80656 100644
--- a/pcrecpp.h
+++ b/pcrecpp.h
@@ -652,6 +652,8 @@ class PCRECPP_EXP_DEFN RE {
  private:
 
   void Init(const string& pattern, const RE_Options* options);
+  // keep the old version from 6.x
+  void Init(const char *pattern, const RE_Options* options);
   void Cleanup();
 
   // Match against "text", filling in "vec" (up to "vecsize" * 2/3) with
